packages/dappmanager/src/modules/installer/https.ts

```diff
@@ -0,0 +1,166 @@
+import { listPackageNoThrow } from "../docker/list/listPackages";
+import { httpsPortal } from "../../calls/httpsPortal";
+import { prettyDnpName } from "../../utils/format";
+import params from "../../params";
+import { InstallPackageData } from "../../types";
+import { Log } from "../../utils/logUi";
+import { HttpsPortalMapping } from "../../common";
+import { getExternalNetworkAliasFromPackage } from "../../domains";
+import { ComposeFileEditor } from "../compose/editor";
+import { dockerListNetworks, dockerCreateNetwork } from "../docker";
+
+/**
+ * Recreate HTTPs portal mapping if installing or updating HTTPs package
+ */
+export async function httpsEnsureNetworkExists(
+  pkg: InstallPackageData,
+  externalNetworkName: string
+): Promise<void> {
+
+  if (pkg.dnpName !== params.HTTPS_PORTAL_DNPNAME) {
+    throw Error(
+      `package must be DNP_HTTPS.`
+    );
+  }
+  const networks = await dockerListNetworks();
+  if (!networks.find(network => network.Name === externalNetworkName)) {
+    await dockerCreateNetwork(externalNetworkName);
+  }
+
+  // Check whether DNP_HTTPS compose has external network persisted
+  const compose = new ComposeFileEditor(pkg.dnpName, pkg.isCore);
+  if (compose.getComposeNetwork(externalNetworkName) === null) {
+    const composeService = compose.firstService();
+    const alias = getExternalNetworkAliasFromPackage(
+      pkg.dnpName,
+      composeService.serviceName
+    );
+    const aliases = [alias];
+    composeService.addNetwork(externalNetworkName, { aliases });
+    compose.write();
+  }
+}
+
+/**
+ * Persist external network on packages compose files
+ */
+export async function httpsPersistPackagesExternalNetwork(
+  pkg: InstallPackageData,
+  externalNetworkName: string
+): Promise<void> {
+
+  if (!(await hasRunningHTTPS())) { // if there is no https, checks aren't needed
+    return;
+  }
+  const compose = new ComposeFileEditor(pkg.dnpName, pkg.isCore);
+  const services = Object.entries(compose.services());
+
+  for (const [serviceName, composeServiceEditor] of services) {
+    if (await httpsPortal.hasMapping(pkg.dnpName, serviceName)) {
+      const networks = await dockerListNetworks();
+      if (!networks.find(network => network.Name === externalNetworkName)) {
+        await dockerCreateNetwork(externalNetworkName);
+      }
+
+      const alias = getExternalNetworkAliasFromPackage(
+        pkg.dnpName,
+        serviceName
+      );
+      const aliases = [alias];
+      composeServiceEditor.addNetwork(externalNetworkName, { aliases });
+      compose.write();
+      break;
```

Comment:
Why is there a break here?

Reply:
We need to add external network once.

Reply:
I'm very confused by the code, could you explain what it's doing?

Reply:
It adds external network to the HTTPS compose file on install/update if a mapping exists. This was cause of issues because mapping would exist, but on update HTTPS wouldn't reconnect to the external network.